src/app/dev/DevToys.Tools/Helpers/ListCompareHelper.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Markdig.Helpers;
+using Microsoft.Extensions.Logging;
+
+namespace DevToys.Tools.Helpers;
+
+internal static partial class ListCompareHelper
+{
+    public static async ValueTask<ResultInfo<string>> CompareAsync(
+        string firstList,
+        string secondList,
+        bool caseSensitive,
+        ListComparisonMode comparisonMode,
+        ILogger logger,
+        CancellationToken cancellationToken)
+    {
+        await TaskSchedulerAwaiter.SwitchOffMainThreadAsync(cancellationToken);
+        ResultInfo<string> compareResult;
+        List<string> listA = new List<string>(firstList.Split(Environment.NewLine));
```

Comment:
Using `Environment.NewLine` is a good start that will probably satisfy the major part of customers. However we're aware that in DevToys 1.0 some people were bothered by the fact that we were only supporting one kind of end of line sequence.
To maximize compatibility, you can use the following helper method that splits a text following `LineFeed` (`\n`), `CarriageReturn` (`\r`) and `CarriageReturnLineFeed` (`\r\n`):
https://github.com/veler/DevToys/blob/6fa986b48351eb4f9a86bacc9617652f465ff8ee/src/app/dev/DevToys.Tools/Helpers/MemoryExtensions.cs#L98-L114

Reply:
Note to myself and to @btiteux : perhaps we should consider moving some of these helpers to `DevToys.API` so extenders won't have to reinvent the wheel when creating extensions (outside of DevToys.Tools). Thoughts?

Reply:
All right for separators

Reply:
Thanks 😊
Looking at it again, I wonder if it's possible to do the `Intersect` and `Except` (or equivalent implementation without using LINQ) using `ReadOnlyMemory<char>`. That would allow us not to use the `Split` method (and to reuse the `ToLines` method I suggested above). The big advantage of it is that we would allocate less memory in scenario where the user input a large text that have very few differences. In comparison, if I enter a large text with the current implementation, whether there are many or zero differences between the 2 lists, the memory allocation will climb as the `Split` allocates a new string for every line it finds.

We are aware there are customers who open/paste very large document in DevToys (see #953 for example). We found that DevToys 1.0 isn't good at it because of excessive use of `ToString`, `Split`, `Substring` and other similar string manipulation method that may result in allocate large objects, which contributes to a high memory usage and hang during the allocation.

PS: that's something we will try to document clearly when we will document the API 😊

Reply:
That's right, can you check if it's ok right now ?
Moreover, i've decided to remove duplicate entries in a list, are you agree with that ?

Reply:
Yes it looks much better, thank you for this! 😁
And yes, I agree with removing duplicate entries. It seems to be consistent with http://www.listdiff.com/